### ISSUE TITLE
Generate relative XPath from element without id/class

### ISSUE THREAD

**Issue by ORESoftware** [johannhof/xpath-dom#3]
Hey! Your library looks like what I need. I am looking to generate XPath strings for elements on the page using events. For example, say I have a MouseEvent target element - perhaps that element does not have an id/class. So I want to generate a relative XPath for that element using the nearest element with an id.

Can I do that with this library? How? I am looking for something like:

```js
const el = e.target;
xpath.getRelativeXPath(el, 'id');
```

what this would do is generate a relative XPath from the nearest element with the id, to el, something like this:

```html
<div id="foo">
     <div><button>click me</button></div>
</div>
```

and the relative Xpath given the above would be something like:

`//div[@id='foo']/div/button`

is that possible with this lib?

**Comment by johannhof** [comment]
You could do something like this using the [Element.closest](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) API (or a polyfill), since the getXPath function allows you to pass an optional root node.

```js
xpath.getXPath(el, el.closest('[id]'));
```

Would that work for you?

**Comment by ORESoftware** [comment]
cool, I am glad you understand my question, so `el.closest()` is a baked-in browser utility? I assume `closest()` goes up the tree, not down the branches.

**Comment by johannhof** [comment]
Yup, goes up to the next parent, see the MDN link :)On Jan 20, 2018 19:59, Operations Research Engineering Software+ <notifications@github.com> wrote:cool, so el.closest() is a baked-in browser utiltity? I assume closest() goes up the tree, not down the branches.

**Comment by ORESoftware** [comment]
ok cool, yeah I had to write some pretty gnarly code to generate proper relative XPaths, it would be nice to have something part of this library that was more refined than what I came up with
